src/plugins/saved_objects_management/public/management_section/recent_work.tsx

```diff
@@ -0,0 +1,276 @@
+});
+
+const recentlyUpdated = i18n.translate('homepage.recentWorkSection.recentlyUpdated', {
+  defaultMessage: 'recently updated',
+});
+
+const sortKeyMap = {
+  [recentlyViewed]: 'lastAccessedTime',
+  [recentlyUpdated]: 'updatedAt',
+} as const;
+
+type KeyOf<T> = keyof T;
+
+function sortBy<T>(key: KeyOf<T>) {
+  return (a: T, b: T): number => (a[key] > b[key] ? -1 : b[key] > a[key] ? 1 : 0);
+}
+
+type DetailedRecentlyAccessedItem = ChromeRecentlyAccessedHistoryItem &
+  SavedObjectWithMetadata &
+  ChromeRecentlyAccessedHistoryItem['meta'] & {
+    updatedAt: number;
+    workspaceName?: string;
+  };
+
+const bulkGetDetail = (
+  savedObjects: Array<Pick<SavedObject, 'type' | 'id'>>,
+  http: CoreStart['http']
+) => {
+  return Promise.all(
+    savedObjects.map((obj) =>
+      http
+        .get<SavedObjectWithMetadata>(
+          `/api/opensearch-dashboards/management/saved_objects/${encodeURIComponent(
+            obj.type
+          )}/${encodeURIComponent(obj.id)}`
+        )
+        .catch((error) => ({
+          id: obj.id,
+          type: obj.type,
+          error,
+          attributes: {},
+          references: [],
+          meta: {},
+          updated_at: '',
+        }))
+    )
+  );
+};
+
+const widthForTypeSelector = 220;
+const widthForRightMargin = 4;
+const MAX_ITEMS_DISPLAY = 6;
+
+export const RecentWork = (props: { core: CoreStart; workspaceEnabled?: boolean }) => {
+  const { core, workspaceEnabled } = props;
+  const recently$Ref = useRef(core.chrome.recentlyAccessed.get$());
+  const recentAccessed = useObservable(recently$Ref.current, []);
+  const workspaceList = useObservable(core.workspaces.workspaceList$, []);
+  const [selectedType, setSelectedType] = useState(allOption);
+  const [selectedSort, setSelectedSort] = useState(recentlyViewed);
+  const [detailedSavedObjects, setDetailedSavedObjects] = useState<DetailedRecentlyAccessedItem[]>(
+    []
+  );
+
+  const allOptions = useMemo(() => {
+    const options: string[] = [allOption];
+    detailedSavedObjects
+      .filter((item) => !item.error)
+      .forEach((recentAccessItem: ChromeRecentlyAccessedHistoryItem) => {
+        if (recentAccessItem.meta?.type && options.indexOf(recentAccessItem.meta.type) === -1) {
+          options.push(recentAccessItem.meta.type);
+        }
+      });
+    return options.map((option: string) => ({ label: option, value: option }));
+  }, [detailedSavedObjects]);
+
+  const itemsForDisplay = useMemo(() => {
+    const sortedResult = [...detailedSavedObjects]
+      .filter((item) => !item.error)
+      .sort(sortBy(sortKeyMap[selectedSort]));
+    return sortedResult.filter((item: SavedObject & ChromeRecentlyAccessedHistoryItem) => {
+      if (selectedType === allOption) return true;
+      return item.type === selectedType;
+    });
+  }, [detailedSavedObjects, selectedSort, selectedType]);
+
+  useEffect(() => {
+    const savedObjects = recentAccessed
+      .filter((item) => item.meta?.type)
```

Comment:
does this mean legacy recently view items are ignored?

Reply:
Yes, because legacy items does not have viewed at, nor the type.